src/compiler/scala/tools/nsc/typechecker/Contexts.scala

```diff
@@ -1777,28 +1774,28 @@ trait Contexts { self: Analyzer =>
       if (record && settings.warnUnusedImport && selectors.nonEmpty && result != NoSymbol && pos != NoPosition)
         recordUsage(current, result)
 
-      // Harden against the fallout from bugs like scala/bug#6745
+      // Harden against the fallout from bugs like scala/bug#6745 and #5389
+      // Enforce no importing universal members from root import Predef modules.
       //
       // [JZ] I considered issuing a devWarning and moving the
       //      check inside the above loop, as I believe that
       //      this always represents a mistake on the part of
       //      the caller.
-      if (definitions isImportable result) result
-      else NoSymbol
-    }
-    private def selectorString(s: ImportSelector): String = {
-      if (s.isWildcard) "_"
-      else if (s.isRename) s.name + " => " + s.rename
-      else "" + s.name
+      result.filter(sym =>
+        if (isRootImport) !definitions.isUnimportableUnlessRenamed(sym)
+        else definitions.isImportable(sym)
+      )
     }
 
     def allImportedSymbols: Iterable[Symbol] =
-      importableMembers(qual.tpe) flatMap (transformImport(tree.selectors, _))
+      importableMembers(qual.tpe).flatMap(transformImport(tree.selectors, _))
 
     @tailrec
     private def transformImport(selectors: List[ImportSelector], sym: Symbol): List[Symbol] = selectors match {
       case Nil => Nil
-      case sel :: Nil if sel.isWildcard => List(sym)
+      case sel :: Nil if sel.isWildcard =>
+        if (isRootImport && definitions.isUnimportableUnlessRenamed(sym)) Nil
```

Comment:
i.e. `isUniversalMember`, but using this method for uniformity with the main case above. `allImportedSymbols` is used for completions.